experiment/cherrypicker/main.go

```diff
@@ -0,0 +1,167 @@
+	githubEndpoint    = flag.String("github-endpoint", "https://api.github.com", "GitHub's API endpoint.")
+	githubTokenFile   = flag.String("github-token-file", "/etc/github/oauth", "Path to the file containing the GitHub OAuth secret.")
+	webhookSecretFile = flag.String("hmac-secret-file", "/etc/webhook/hmac", "Path to the file containing the GitHub HMAC secret.")
+	// TODO: Read from the plugin config in prow
+	r = flag.String("repo", "", "Repository for doing cherry-picks, eg. kubernetes/test-infra.")
+)
+
+func main() {
+	flag.Parse()
+	logrus.SetFormatter(&logrus.JSONFormatter{})
+	// Ignore SIGTERM so that we don't drop hooks when the pod is removed.
+	// We'll get SIGTERM first and then SIGKILL after our graceful termination
+	// deadline.
+	signal.Ignore(syscall.SIGTERM)
+
+	org, repo, err := validateRepo(*r)
+	if err != nil {
+		logrus.WithError(err).Fatal("Invalid --repo value.")
+	}
+
+	webhookSecretRaw, err := ioutil.ReadFile(*webhookSecretFile)
+	if err != nil {
+		logrus.WithError(err).Fatal("Could not read webhook secret file.")
+	}
+	webhookSecret := bytes.TrimSpace(webhookSecretRaw)
+
+	oauthSecretRaw, err := ioutil.ReadFile(*githubTokenFile)
+	if err != nil {
+		logrus.WithError(err).Fatal("Could not read oauth secret file.")
+	}
+	oauthSecret := string(bytes.TrimSpace(oauthSecretRaw))
+
+	_, err = url.Parse(*githubEndpoint)
+	if err != nil {
+		logrus.WithError(err).Fatal("Must specify a valid --github-endpoint URL.")
+	}
+
+	githubClient := github.NewClient(oauthSecret, *githubEndpoint)
+	if *dryRun {
+		githubClient = github.NewDryRunClient(oauthSecret, *githubEndpoint)
+	}
+
+	gitClient, err := git.NewClient()
+	if err != nil {
+		logrus.WithError(err).Fatal("Error getting git client.")
+	}
+
+	logger := logrus.StandardLogger()
+	// TODO: Use global option from the prow config.
+	logrus.SetLevel(logrus.DebugLevel)
+	githubClient.Logger = logger.WithField("client", "github")
+	gitClient.Logger = logger.WithField("client", "git")
+
+	// The bot name is used to determine to what fork we can push cherry-pick branches.
+	botName, err := githubClient.BotName()
+	if err != nil {
+		logrus.WithError(err).Fatal("Error getting bot name.")
+	}
+
+	repos, err := githubClient.GetRepos(botName, true)
+	if err != nil {
+		logrus.WithError(err).Fatal("Error listing bot repositories.")
+	}
+	var exists bool
+	for _, r := range repos {
+		if !r.Fork {
+			continue
+		}
+		if r.FullName == botName+"/"+repo {
+			exists = true
+			break
+		}
+	}
+	// Fork repo if it doesn't exist.
+	if !exists {
+		if err := githubClient.CreateFork(org, repo); err != nil {
+			logrus.WithError(err).Fatalf("Error while forking %s/%s.", org, repo)
+		}
+		// Wait for at most 5 minutes for the fork to appear on Github.
+		after := time.After(5 * time.Minute)
+		tick := time.Tick(5 * time.Second)
+		forkURL := fmt.Sprintf("https://github.com/%s/%s", botName, repo)
+	out:
```

Comment:
May be easier to read with some functions instead of the label and four-times indentation

Reply:
done